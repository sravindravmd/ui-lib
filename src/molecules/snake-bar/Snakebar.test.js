import React from 'react'
import { SnakeBar } from './'
import { render, cleanup } from '@testing-library/react'
import 'regenerator-runtime/runtime'

afterEach(cleanup)

const successProp = {
  alertIcon: 'icon-success',
  dismissable: true,
  kind: 'success',
  label: 'Image added succesfully',
  show: true,
  withIcon: true,
}

const alertProp = {
  alertIcon: 'icon-success',
  dismissable: true,
  kind: 'success',
  label: 'Something went wrong. Please try again',
  show: true,
  withIcon: true,
}

describe('<Snake Bar/>', () => {
  it('renders success ui', () => {
    const { container } = render(<SnakeBar {...successProp} />)
    expect(container).toMatchSnapshot()
  })

  it('renders alert ui', () => {
    const { container } = render(<SnakeBar {...alertProp} />)
    expect(container).toMatchSnapshot()
  })
})
