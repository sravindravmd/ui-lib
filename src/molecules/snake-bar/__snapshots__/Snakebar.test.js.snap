// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Snake Bar/> renders alert ui 1`] = `
<div>
  <div
    class="sc-bdfBwQ eTnYbJ success "
  >
    <div
      class="row"
    >
      <div
        class="icon"
      >
        <svg
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M17.912 10.0674C18.2049 9.77455 18.2049 9.29967 17.912 9.00678C17.6191 8.71389 17.1443 8.71389 16.8514 9.00678L10.7919 15.0663L8.0274 12.3018C7.73451 12.009 7.25963 12.009 6.96674 12.3018C6.67385 12.5947 6.67385 13.0696 6.96674 13.3625L10.2421 16.6378C10.2483 16.6445 10.2547 16.6511 10.2612 16.6576C10.4174 16.8139 10.6255 16.8868 10.8301 16.8763C11.0092 16.8673 11.1857 16.7944 11.3225 16.6576C11.3287 16.6514 11.3348 16.6451 11.3407 16.6388L17.912 10.0674Z"
            fill="#008A1E"
            fill-rule="evenodd"
          />
          <circle
            cx="12"
            cy="12"
            r="11"
            stroke="#008A1E"
          />
        </svg>
      </div>
      <div
        class="col col-middle"
      >
        <div>
          <span
            class="label"
          >
            Something went wrong. Please try again
          </span>
        </div>
      </div>
      <div
        class="dismiss"
      >
        <svg
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.5195 4.54108C12.8064 4.2423 12.8064 3.75789 12.5195 3.45912C12.2326 3.16034 11.7674 3.16034 11.4805 3.45912L12.5195 4.54108ZM7.69063 7.40602C7.40373 7.70479 7.40373 8.1892 7.69062 8.48798C7.97752 8.78675 8.44266 8.78675 8.72955 8.48798L7.69063 7.40602ZM11.4805 3.45912L7.69063 7.40602L8.72955 8.48798L12.5195 4.54108L11.4805 3.45912Z"
          />
          <path
            d="M11.4326 12.541C11.7195 12.8398 12.1846 12.8398 12.4715 12.541C12.7584 12.2422 12.7584 11.7578 12.4715 11.459L11.4326 12.541ZM8.68166 7.51208C8.39477 7.2133 7.92963 7.2133 7.64273 7.51207C7.35584 7.81084 7.35583 8.29525 7.64272 8.59403L8.68166 7.51208ZM12.4715 11.459L8.68166 7.51208L7.64272 8.59403L11.4326 12.541L12.4715 11.459Z"
          />
          <path
            d="M3.48151 11.4589C3.19461 11.7577 3.19461 12.2421 3.48151 12.5409C3.7684 12.8397 4.23354 12.8397 4.52043 12.5409L3.48151 11.4589ZM8.31035 8.59398C8.59724 8.29521 8.59724 7.8108 8.31035 7.51202C8.02346 7.21325 7.55832 7.21325 7.27143 7.51202L8.31035 8.59398ZM4.52043 12.5409L8.31035 8.59398L7.27143 7.51202L3.48151 11.4589L4.52043 12.5409Z"
          />
          <path
            d="M4.56839 3.45902C4.2815 3.16024 3.81636 3.16023 3.52947 3.459C3.24257 3.75778 3.24256 4.24218 3.52945 4.54096L4.56839 3.45902ZM7.31932 8.48792C7.60621 8.7867 8.07135 8.7867 8.35824 8.48793C8.64514 8.18916 8.64515 7.70475 8.35826 7.40597L7.31932 8.48792ZM3.52945 4.54096L7.31932 8.48792L8.35826 7.40597L4.56839 3.45902L3.52945 4.54096Z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Snake Bar/> renders success ui 1`] = `
<div>
  <div
    class="sc-bdfBwQ eTnYbJ success "
  >
    <div
      class="row"
    >
      <div
        class="icon"
      >
        <svg
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M17.912 10.0674C18.2049 9.77455 18.2049 9.29967 17.912 9.00678C17.6191 8.71389 17.1443 8.71389 16.8514 9.00678L10.7919 15.0663L8.0274 12.3018C7.73451 12.009 7.25963 12.009 6.96674 12.3018C6.67385 12.5947 6.67385 13.0696 6.96674 13.3625L10.2421 16.6378C10.2483 16.6445 10.2547 16.6511 10.2612 16.6576C10.4174 16.8139 10.6255 16.8868 10.8301 16.8763C11.0092 16.8673 11.1857 16.7944 11.3225 16.6576C11.3287 16.6514 11.3348 16.6451 11.3407 16.6388L17.912 10.0674Z"
            fill="#008A1E"
            fill-rule="evenodd"
          />
          <circle
            cx="12"
            cy="12"
            r="11"
            stroke="#008A1E"
          />
        </svg>
      </div>
      <div
        class="col col-middle"
      >
        <div>
          <span
            class="label"
          >
            Image added succesfully
          </span>
        </div>
      </div>
      <div
        class="dismiss"
      >
        <svg
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.5195 4.54108C12.8064 4.2423 12.8064 3.75789 12.5195 3.45912C12.2326 3.16034 11.7674 3.16034 11.4805 3.45912L12.5195 4.54108ZM7.69063 7.40602C7.40373 7.70479 7.40373 8.1892 7.69062 8.48798C7.97752 8.78675 8.44266 8.78675 8.72955 8.48798L7.69063 7.40602ZM11.4805 3.45912L7.69063 7.40602L8.72955 8.48798L12.5195 4.54108L11.4805 3.45912Z"
          />
          <path
            d="M11.4326 12.541C11.7195 12.8398 12.1846 12.8398 12.4715 12.541C12.7584 12.2422 12.7584 11.7578 12.4715 11.459L11.4326 12.541ZM8.68166 7.51208C8.39477 7.2133 7.92963 7.2133 7.64273 7.51207C7.35584 7.81084 7.35583 8.29525 7.64272 8.59403L8.68166 7.51208ZM12.4715 11.459L8.68166 7.51208L7.64272 8.59403L11.4326 12.541L12.4715 11.459Z"
          />
          <path
            d="M3.48151 11.4589C3.19461 11.7577 3.19461 12.2421 3.48151 12.5409C3.7684 12.8397 4.23354 12.8397 4.52043 12.5409L3.48151 11.4589ZM8.31035 8.59398C8.59724 8.29521 8.59724 7.8108 8.31035 7.51202C8.02346 7.21325 7.55832 7.21325 7.27143 7.51202L8.31035 8.59398ZM4.52043 12.5409L8.31035 8.59398L7.27143 7.51202L3.48151 11.4589L4.52043 12.5409Z"
          />
          <path
            d="M4.56839 3.45902C4.2815 3.16024 3.81636 3.16023 3.52947 3.459C3.24257 3.75778 3.24256 4.24218 3.52945 4.54096L4.56839 3.45902ZM7.31932 8.48792C7.60621 8.7867 8.07135 8.7867 8.35824 8.48793C8.64514 8.18916 8.64515 7.70475 8.35826 7.40597L7.31932 8.48792ZM3.52945 4.54096L7.31932 8.48792L8.35826 7.40597L4.56839 3.45902L3.52945 4.54096Z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
